velox/type/DecimalUtil.h

```diff
@@ -80,31 +80,30 @@ class DecimalUtil {
     }
   }
 
-  template <typename TOutput>
-  inline static std::optional<TOutput> rescaleBigint(
+  template <TypeKind InputKind, typename TOutput>
```

Comment:
Does this really need to be a template variable ? It seems like we are only using it to print the type name in an error.

Reply:
Agree, I am just restricting checks to compile time. If we pass as a param, then it will be a runtime check. Also, I am asserting that this method is templatized only for ints. This will reduce duplicate codes.

Reply:
For this case, you can get the TypeKind from the CPP type.

Reply:
You have to make this `template <typename TInput, typename TOuput>`